<?xml version="1.0"?>
<launch>
  <arg name="side" default="r"/> <!-- "b" = blue side, "r" = red side  -->

    <!-- You can change control program  -->
    <!-- make args to params -->
    <group if="$(eval side=='r')">
      <param name="/my_side"    value="r"/>
      <param name="/enemy_side" value="b"/>
    </group>
    <group if="$(eval side=='b')">
      <param name="/my_side" value="b"/>
      <param name="/enemy_side" value="r"/>
    </group>

    <!-- main program -->
    <node pkg="burger_war" type="burger_war_main_final.py" name="burger_war_main" output="screen"/>

    <!-- navigation -->
    <include file="$(find burger_navigation)/launch/burger_navigation.launch"/>

    <!-- /scan based enemy detection -->
    <include file="$(find burger_war)/launch/obstacle_detector.launch"/>
    <node pkg="burger_war" type="enemy_detector.py" name="enemy_detector_node" output="screen"/>

    <!-- remove noise(ex. transparent pole) from a /scan -->
    <node pkg="burger_war" type="scan_trimer.py" name="scan_trimer_node" output="screen"/>

    <!-- observe the maker's state and publish it -->
    <node pkg="burger_war" type="targets_publisher.py" name="marker_status_publisher_node" output="screen"/>

    <!-- Rviz -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find burger_war)/rviz/rviz_config.rviz"/>
    
</launch>

